### PR TITLE
Theme: remove nested anchor in theme sheet

### LINKED ISSUE #34609: Theme: nested anchor tags in ThemeSheet.renderScreenshot
#### Steps to reproduce

1. Starting at https://wordpress.com/theme/photo-blog in Chrome.
2. Open Chrome DevTools.
3. In Console tab, there should be a warning:

screenshot_776

4. Switch to Element tab and search for "theme__sheet-screenshot".

screenshot_775

#### Context / Source

In `client/my-sites/theme/main.jsx` file, `renderScreenshot()` wraps call to `renderPreviewButton()` with an anchor.

### PR DESCRIPTION
**Pull request by donpark**
#### Changes proposed in this Pull Request

Removes unnecessary 'Open Live Demo' anchor nested inside preview anchor in theme sheet page at `/theme/{theme_slug}` path.

Nested anchor was not visible and is no longer necessary because 'Open Live Demo' button is already displayed in the Nav Bar.

screenshot_790

#### Testing instructions

1. Using Chrome, navigate to `/theme/photo-blog` or `/theme/photo-blog/{site_slug}`.
2. Open DevTools > Console and verify that following warnings are no longer emitted.

screenshot_789

*

Fixes #34609

### CODE REVIEW

File: client/my-sites/theme/main.jsx [hunk 1]
```diff
@@ -228,24 +228,6 @@ class ThemeSheet extends React.Component {
...
-				rel="noopener noreferrer"
-			>
-				<span className="theme__sheet-preview-link-text">
-					{ i18n.translate( 'Open Live Demo', {
-						context: 'Individual theme live preview button',
-					} ) }
-				</span>
-			</a>
-		);
-	}
```
Comment (jsnajdr) [inline client/my-sites/theme/main.jsx]: This "Open Live Demo" button seems to be a part of intentional UI design that looks differently on wide two-column layout and narrow, mobile one-column layout
- two-column layout shows an "Open Live Demo" item in a `SectionNav` in the left column
- that item is hidden in one-column view, and the "Open Live Demo" button (the element being removed in this PR) is displayed under the screenshot

@folletto may provide some advice about how to remove the invalid HTML while keeping the design working. He originally worked on it and submitted PRs like #13598.

Also, removing the button breaks the `theme-sheet-welcome-tour` guided tour that looks for the `data-tip-target="theme-sheet-preview"` element and attaches a bubble to it. That tour is likely already broken on non-mobile views (?!)

Comment (jsnajdr) [reply client/my-sites/theme/main.jsx]: To see this code in action, open a theme page like https://wordpress.com/theme/modern-business and resize the window to switch between one- and two-column views.

**Comment by folletto** [comment]
> because 'Open Live Demo' button is already displayed in the Nav Bar

Maybe I'm missing something, but... where? Can you post a screenshot?

**Comment by jsnajdr** [comment]
Two column layout shows "Open Live Demo" in the `SectionNav`. The right column contains the theme screenshot and nothing else. The "Open Live Button" is not displayed there:

Screenshot 2019-07-17 at 17 12 46

Single column layout hides "Open Live Demo" from the `SectionNav` and shows it as button in an overlay over the screenshot. (The responsive view shows the second column on top, and the first column under it)
Screenshot 2019-07-17 at 17 13 18

**Comment by folletto** [comment]
Yes sorry, I didn't mean the difference between desktop and mobile.

I thought that quote in the original post meant "the action already exists elsewhere on the page" but it seems that if it gets removed, then there's no explicit action for it anymore: "no longer necessary".

File: client/my-sites/theme/main.jsx (hunk 1 above)
Comment (donpark) [reply client/my-sites/theme/main.jsx]: Ah, so that's why it was there. We should include a comment in the fix so it's more apparent.

**Comment by donpark** [comment]
I'm not sure if current implementation of NavSection supports inlined right-alined button but I think we can improve current design of the theme sheet page by turning "Open Live Demo" into a button that remains in-place even in mobile view. As to why, it's not really a NavItem but a button and it could use more emphasis.

This is what  I had in mind. Feel free to ping relevant folks for design review on this:

screenshot_795

**Comment by jsnajdr** [comment]
@folletto we use a similar layout on the "Plugins" screen:

Screenshot 2019-07-18 at 11 50 11

But the buttons are not part of the `SectionNav`, and they currently don't align very well when the screen is resized to a narrow width.

@donpark If we want to just fix the non-valid HTML markup (nested `a`'s), then we can just convert the "Open Live Demo" button to a passive `div`, can't we? It's already part of the screenshot's clickable area, so it doesn't need to be an active element with `href`s or event handlers.

**Comment by folletto** [comment]
> we use a similar layout on the "Plugins" screen

That's a bit different from the above. I'm ok with going for something like Plugins if y'all prefer to go for that layout :)

I agree right now is probably better to just fix the nesting.

**Comment by donpark** [comment]
>  If we want to just fix the non-valid HTML markup (nested a's), then we can just convert the "Open Live Demo" button to a passive div, can't we? It's already part of the screenshot's clickable area, so it doesn't need to be an active element with hrefs or event handlers.

Yeah, I agree that just fixing the nesting makes more sense. I'll update this PR. Thx.

**Comment by donpark** [comment]
Anchor has been replaced with `<div>`. Pseudo-button is still visible and clickable as before in mobile view:

screenshot_797

**Comment by folletto** [comment]
I just noticed a thing on the mobile screenshot: the gradient is wrong. :D 

It's not super noticeable on that theme given the image cut, but on another:

Screenshot 2019-07-18 at 23 52 48


This is fixed:

Screenshot 2019-07-18 at 23 54 33



It might have been when the color variables were introduced, this slipped through.

```linear-gradient(180deg,rgba(var(--color-neutral-0-rgb),1) 0,rgba(var(--color-neutral-0-rgb),.5) 40%,var(--color-neutral-0) 93%)```

This should fix it (but I haven't checked if it has consequences elsewhere):

```linear-gradient(180deg,transparent 0,rgba(var(--color-neutral-0-rgb),.5) 40%,var(--color-neutral-0) 93%)```

**Comment by donpark** [comment]
> This should fix it (but I haven't checked if it has consequences elsewhere):
> 
> `linear-gradient(180deg,transparent 0,rgba(var(--color-neutral-0-rgb),.5) 40%,var(--color-neutral-0) 93%)`

Done. Button border looked more vulnerable as result but darkening it didn't help much so I let it alone.

**Pull request merged**